apps/python-service/dependencies.py

```
@@ -0,0 +1,48 @@
+
+async def verify_signature(request: Request):
+    signature = request.headers.get("X-Internal-Signature")
+    timestamp_str = request.headers.get("X-Timestamp")
+
+    if not signature or not timestamp_str:
+        raise HTTPException(status_code=403, detail="Missing authentication headers")
+
+    try:
+        timestamp = int(timestamp_str)
+    except ValueError:
+        raise HTTPException(status_code=403, detail="Invalid timestamp format")
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)